classes/helper/datafield_manager.php

```diff
@@ -83,11 +83,17 @@ public function get_datafields($event = null, $stepresults = null) {
      * the workflow.
      */
     public function update_datafields($event, $stepresults) {
+        global $CFG;
+        $eventvaribales = [];
         $newfields = [];
         if (is_array($event->get_data())) {
-            $newfields = array_merge($newfields, $event->get_data());
+            $eventvaribales = $event->get_data();
+            $eventvaribales['wwwroot'] = isset($eventvaribales['wwwroot']) ? $eventvaribales['wwwroot'] : $CFG->wwwroot;
+            $newfields = array_merge($newfields, $eventvaribales);
         }
         if (is_array($event->get_logextra())) {
+            $eventvaribales = $event->get_logextra();
+            $eventvaribales['wwwroot'] = isset($eventvaribales['wwwroot']) ? $eventvaribales['wwwroot'] : $CFG->wwwroot;
```

Comment:
I would just force set this without the isset, and do it about 10 lines lower outside the loops so it is just always wwwroot, and can never be overwritten by other data